WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/ProductFormActionsFactory+VisibilityTests.swift

```diff
@@ -160,6 +160,36 @@ final class ProductFormActionsFactory_VisibilityTests: XCTestCase {
         XCTAssertFalse(factory.settingsSectionActions().contains(.briefDescription))
         XCTAssertTrue(factory.bottomSheetActions().contains(.editBriefDescription))
     }
+
+    // MARK: - Downloadable Files
+
+    func testDownloadableFilesRowIsVisibleForDownloadableProductWithNonEmptyDownloadableFiles() {
+        // Arrange
+        let product = Fixtures.downloadableProduct
+        let model = EditableProductModel(product: product)
+
+        // Action
+        let factory = ProductFormActionsFactory(product: model,
+                                                formType: .edit,
+                                                isEditProductsRelease3Enabled: true)
+
+        // Assert
+        XCTAssertTrue(factory.settingsSectionActions().contains(.downloadableFiles))
+    }
+
+    func testDownloadableFilesRowIsInvisibleForNonDownloadableProductWithoutDownloadableFiles() {
+        // Arrange
+        let product = Fixtures.nonDownloadableProduct
+        let model = EditableProductModel(product: product)
+
+        // Action
+        let factory = ProductFormActionsFactory(product: model,
+                                                formType: .edit,
+                                                isEditProductsRelease3Enabled: true)
+
+        // Assert
+        XCTAssertFalse(factory.settingsSectionActions().contains(.downloadableFiles))
+    }
```

Comment:
It seems that you are not considering the feature flag M5 here. It should be included in the `ProductFormActionsFactory` init for testing purposes.

Reply:
@pmusolino , I am a bit confused here about what would be best. 
I see `isEditProductsRelease3Enabled`  value on `ProductFormActionsFactory`  is being cascaded from `AppSettingsStore` ’s `loadProductsFeatureSwitch` . Do you think I should do something like that?
Or adding a constructor parameter like this `isEditProductsRelease5Enabled: Bool = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.editProductsRelease5)`  on `ProductFormActionsFactory`  would be enough. Please advise.

<img width="584" alt="Screen Shot 2020-09-25 at 6 23 45 PM" src="https://user-images.githubusercontent.com/1577598/94268043-65e90480-ff5e-11ea-9ab9-a49c1c8f4986.png">